pkg/ir/ir.go

```diff
@@ -0,0 +1,66 @@
+package ir
+
+import (
+	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+// Xds holds the intermediate representation of a Gateway and is
+// used by the xDS Translator to convert it into xDS resources
+type Xds struct {
+	Name string
+	// One or more HTTP or HTTPS listeners exposed by the gateway
+	Http []HttpListener
+}
+
+type HttpListener struct {
```

Comment:
Why doesn't `HttpListener` have an address field that represents an IPv4/IPv6 address?

Reply:
I actually think we may need to put that in the parent structure maybe? In the Gateway API, it's possible to describe:
- listeners bound to no specific address (that is, the data plane should just figure it out)
- one or more listeners bound to one or more addresses.

So, we will need to be able to describe "this listener or these listeners should be attached to this address or these addresses" in the IR somehow.

Reply:
I added `Address` into `HttpListener` for now, both the approaches of moving address+port to top level or within listener spec all depend on what Gateway API translator prefers when breaking up the cases referred by 
@youngnick, will defer to @skriss for his preference since he is designing the GW API translator